diff -urN aa/gimp-2.0rc1/app/paint-funcs/paint-funcs.h gimp-2.0rc1/app/paint-funcs/paint-funcs.h
--- aa/gimp-2.0rc1/app/paint-funcs/paint-funcs.h	2004-03-10 14:13:43.000000000 +0100
+++ gimp-2.0rc1/app/paint-funcs/paint-funcs.h	2004-03-21 14:44:37.713167800 +0100
@@ -356,7 +356,7 @@
 					   gint,
                                            GimpLayerModeEffects,
                                            gboolean *,
-                                           CombinationMode);
+                                           InitialMode);
 
 void  combine_regions                     (PixelRegion *, PixelRegion *,
 					   PixelRegion *, PixelRegion *,
