--- gimp-2.10.38/plug-ins/file-tiff/file-tiff-load.c.orig	2024-05-03 02:33:35.000000000 +0200
+++ gimp-2.10.38/plug-ins/file-tiff/file-tiff-load.c	2024-09-05 20:59:10.593974143 +0200
@@ -1301,8 +1301,8 @@ load_image (GFile        *file,
 
       /* any resolution info in the file? */
       {
-        gfloat   xres = 72.0;
-        gfloat   yres = 72.0;
+        gdouble  xres = 72.0;
+        gdouble  yres = 72.0;
         gushort  read_unit;
         GimpUnit unit = GIMP_UNIT_PIXEL; /* invalid unit */
 
--- gimp-2.10.38/plug-ins/metadata/metadata-editor.c.orig	2024-05-03 02:33:35.000000000 +0200
+++ gimp-2.10.38/plug-ins/metadata/metadata-editor.c	2024-09-05 21:43:47.976168927 +0200
@@ -2140,7 +2140,7 @@ metadata_dialog_editor_set_metadata (GEx
                             }
                           else
                             {
-                              if (! g_strv_contains (values, equiv_values[evi]))
+                              if (! g_strv_contains ((const gchar * const *)values, equiv_values[evi]))
                                 {
                                   gchar *tmpvalue;
 
