diff -uNr gimp-1.2.1.orig/plug-ins/script-fu/script-fu-intl.h gimp-1.2.1/plug-ins/script-fu/script-fu-intl.h
--- gimp-1.2.1.orig/plug-ins/script-fu/script-fu-intl.h	Sun Feb 27 21:45:58 2000
+++ gimp-1.2.1/plug-ins/script-fu/script-fu-intl.h	Wed Jun 13 19:03:32 2001
@@ -25,7 +25,7 @@
 
 #ifdef HAVE_LC_MESSAGES
 #define INIT_I18N()	G_STMT_START{			\
-  setlocale(LC_MESSAGES, ""); 				\
+  setlocale(LC_ALL, ""); 				\
   bindtextdomain("gimp-libgimp", LOCALEDIR);            \
   bindtextdomain("gimp-script-fu", LOCALEDIR);	        \
   textdomain("gimp-script-fu");			        \
@@ -40,7 +40,7 @@
 
 #define INIT_I18N_UI()	G_STMT_START{	\
   gtk_set_locale();			\
-  setlocale (LC_NUMERIC, "C");		\
+  setlocale (LC_ALL, "C");		\
   INIT_I18N();				\
 			}G_STMT_END
 
