--- gimp-2.7.1/libgimpbase/gimpenv.c~	2010-06-13 21:15:08.000000000 +0200
+++ gimp-2.7.1/libgimpbase/gimpenv.c	2010-07-19 10:30:13.787427066 +0200
@@ -259,7 +259,9 @@ gimp_directory (void)
         }
       else
         {
-          const gchar *home_dir = g_get_home_dir ();
+          const gchar *home_dir = g_getenv("HOME_ETC");
+          if (!home_dir)
+            home_dir = g_get_home_dir ();
 
           if (home_dir)
             gimp_dir = g_build_filename (home_dir, env_gimp_dir, NULL);
    {
