--- gimp-2.2.11/data/misc/gimp.desktop.in.in.orig	2006-01-23 14:45:39.000000000 +0000
+++ gimp-2.2.11/data/misc/gimp.desktop.in.in	2006-04-17 10:15:35.000000000 +0100
@@ -9,7 +9,7 @@
 TryExec=gimp-@GIMP_APP_VERSION@
 Icon=@gimpdatadir@/images/@GIMP_DESKTOP_ICON@
 Terminal=false
-Categories=Application;Graphics;2DGraphics;RasterGraphics;GTK;
+Categories=GTK;GNOME;Graphics;RasterGraphics;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=GIMP
 StartupNotify=true
