diff -Nur gimp-1.2.2.orig/data/brushes/Makefile.am gimp-1.2.2/data/brushes/Makefile.am
--- gimp-1.2.2.orig/data/brushes/Makefile.am	Sat Apr  7 13:24:36 2001
+++ gimp-1.2.2/data/brushes/Makefile.am	Fri Aug  3 22:44:15 2001
@@ -59,7 +59,3 @@
 	SketchBrush-32.gih	\
 	SketchBrush-64.gih	\
 	xcf.gbr
-
-install-data-local:
-	rm -f $(brushdatadir)/round1.vbr $(brushdatadir)/pepper.gpb
-
diff -Nur gimp-1.2.2.orig/data/brushes/Makefile.in gimp-1.2.2/data/brushes/Makefile.in
--- gimp-1.2.2.orig/data/brushes/Makefile.in	Thu Jul 26 21:37:17 2001
+++ gimp-1.2.2/data/brushes/Makefile.in	Fri Aug  3 22:44:15 2001
@@ -313,9 +313,6 @@
 
 .PHONY: files
 
-install-data-local:
-	rm -f $(brushdatadir)/round1.vbr $(brushdatadir)/pepper.gpb
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Nur gimp-1.2.2.orig/plug-ins/perl/Makefile.PL gimp-1.2.2/plug-ins/perl/Makefile.PL
--- gimp-1.2.2.orig/plug-ins/perl/Makefile.PL	Thu Jul 26 22:17:43 2001
+++ gimp-1.2.2/plug-ins/perl/Makefile.PL	Fri Aug  3 22:44:15 2001
@@ -233,7 +233,7 @@
 	\$(CHMOD) 755 * ; \\
 	\$(MY_FIXIN) * ; \\
 	for plugin in * ; do \\
-	   $GT \"\$\$plugin\" $GT2 ; \\
+	   $GT \"\$\$plugin\" \$(DESTDIR)$GT2 ; \\
 	done
 	\$(RM_RF) inst-temp
 
diff -Nur gimp-1.2.2.orig/plug-ins/perl/po/Makefile.in.in gimp-1.2.2/plug-ins/perl/po/Makefile.in.in
--- gimp-1.2.2.orig/plug-ins/perl/po/Makefile.in.in	Tue Jun 19 23:51:04 2001
+++ gimp-1.2.2/plug-ins/perl/po/Makefile.in.in	Fri Aug  3 22:44:15 2001
@@ -121,7 +121,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
@@ -223,6 +223,9 @@
 	  fi; \
 	done
 
+# Temporary workaround for building under PLD GNU/Linux
+update-gmo:
+
 POTFILES: POTFILES.in
 	( if test 'x$(srcdir)' != 'x.'; then \
 	    posrcprefix='$(top_srcdir)/'; \
diff -Nur gimp-1.2.2.orig/po-libgimp/Makefile.in.in gimp-1.2.2/po-libgimp/Makefile.in.in
--- gimp-1.2.2.orig/po-libgimp/Makefile.in.in	Thu Jan 27 11:46:58 2000
+++ gimp-1.2.2/po-libgimp/Makefile.in.in	Fri Aug  3 22:44:15 2001
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
diff -Nur gimp-1.2.2.orig/po-script-fu/Makefile.in.in gimp-1.2.2/po-script-fu/Makefile.in.in
--- gimp-1.2.2.orig/po-script-fu/Makefile.in.in	Sat Jun 24 12:46:57 2000
+++ gimp-1.2.2/po-script-fu/Makefile.in.in	Fri Aug  3 22:44:15 2001
@@ -130,7 +130,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
--- gimp-1.2.3/gimptool-1.2.in~	Tue Feb 12 09:58:38 2002
+++ gimp-1.2.3/gimptool-1.2.in	Tue Feb 12 09:58:38 2002
@@ -228,27 +228,27 @@
       case $1 in
         --*install-bin)
           install_cmd="@INSTALL_PROGRAM@"
-          install_dir="$HOME/@gimpdir@/plug-ins"
+          install_dir="${DESTDIR}${HOME}/@gimpdir@/plug-ins"
           ;;
         --install-bin-strip)
           install_cmd="@INSTALL_PROGRAM@ -s"
-          install_dir="$HOME/@gimpdir@/plug-ins"
+          install_dir="${DESTDIR}${HOME}/@gimpdir@/plug-ins"
           ;;
         --*install-admin-bin)
           install_cmd="@INSTALL_PROGRAM@"
-          install_dir="$gimpplugindir/plug-ins"
+          install_dir="${DESTDIR}$(gimpplugindir)/plug-ins"
           ;;
         --install-admin-bin-strip)
           install_cmd="@INSTALL_PROGRAM@ -s"
-          install_dir="$gimpplugindir/plug-ins"
+          install_dir="${DESTDIR}${gimpplugindir}/plug-ins"
           ;;
         --*install-script)
           install_cmd="@INSTALL_DATA@"
-          install_dir="$HOME/@gimpdir@/scripts"
+          install_dir="${DESTDIR}${HOME}/@gimpdir@/scripts"
           ;;
         --*install-admin-script)
           install_cmd="@INSTALL_DATA@"
-          install_dir="$gimpdatadir/scripts"
+          install_dir="${DESTDIR}${gimpdatadir}/scripts"
           ;;
       esac
       case $1 in
@@ -256,7 +256,7 @@
           shift
           if test "x$1" != "x"; then
             if test -f "$install_dir/$1"; then
-              cmd="rm -f $DESTDIR$install_dir/$1"
+              cmd="rm -f ${DESTDIR}${install_dir}/$1"
               test $quiet = "yes" || echo $cmd
               test $donothing = "yes" || exec $cmd
             else
@@ -300,9 +300,9 @@
         --build | --build-noui | --build-nogimpui)
           install_dir=. ;;
         --install | --install-noui | --install-nogimpui)
-          install_dir="$HOME/@gimpdir@/plug-ins" ;;
+          install_dir="${DESTDIR}${HOME}/@gimpdir@/plug-ins" ;;
         --install-admin | --install-admin-noui | --install-admin-nogimpui)
-          install_dir="$gimpplugindir/plug-ins" ;;
+          install_dir="${DESTDIR}${gimpplugindir}/plug-ins" ;;
       esac
       noui=`echo $opt | sed 's/^.*\(noui\)$/\1/'`
       nogimpui=`echo $opt | sed 's/^.*\(nogimpui\)$/\1/'`
