--- gimp-1.3.23/m4macros/gimp-2.0.m4.orig	2003-11-02 23:59:50.000000000 +0100
+++ gimp-1.3.23/m4macros/gimp-2.0.m4	2003-12-28 22:38:23.099012760 +0100
@@ -5,7 +5,7 @@
 dnl AM_PATH_GIMP_2_0([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for GIMP, and define GIMP_CFLAGS and GIMP_LIBS
 dnl
-AC_DEFUN(AM_PATH_GIMP_2_0,
+AC_DEFUN([AM_PATH_GIMP_2_0],
 [dnl 
 dnl Get the cflags and libraries from pkg-config
 dnl
