--- gimp-1.1.13/plug-ins/perl/Gimp/Module.pm.dep	Fri Nov 19 23:21:31 1999
+++ gimp-1.1.13/plug-ins/perl/Gimp/Module.pm	Sun Dec  5 01:44:45 1999
@@ -55,3 +55,5 @@
 =head1 AUTHOR
 
 Marc Lehmann <pcg@goof.com>
+
+=cut
--- gimp-1.1.15.pius/plug-ins/perl/Gimp/Util.pm	Thu Jan  6 00:42:19 2000
+++ gimp-1.1.15/plug-ins/perl/Gimp/Util.pm	Sun Jan 16 22:10:29 2000
@@ -370,3 +370,4 @@
 of the Gimp-Perl extension (contact him to include new functions) is Marc
 Lehmann <pcg@goof.com>
 
+=cut
--- gimp-1.1.15.pius/plug-ins/perl/UI/basewidget.pm	Thu Jan  6 00:42:20 2000
+++ gimp-1.1.15/plug-ins/perl/UI/basewidget.pm	Sun Jan 16 22:10:16 2000
@@ -104,3 +104,4 @@
 
 perl(1), L<Gimp>, L<Gimp::UI>, L<Gtk>.
 
+=cut
