diff -aurN gimp-2.2.11.orig/data/misc/gimp.desktop.in.in gimp-2.2.11/data/misc/gimp.desktop.in.in
--- gimp-2.2.11.orig/data/misc/gimp.desktop.in.in	2006-01-23 15:45:39.000000000 +0100
+++ gimp-2.2.11/data/misc/gimp.desktop.in.in	2006-04-14 10:34:56.000000000 +0200
@@ -9,7 +9,7 @@
 TryExec=gimp-@GIMP_APP_VERSION@
 Icon=@gimpdatadir@/images/@GIMP_DESKTOP_ICON@
 Terminal=false
-Categories=Application;Graphics;2DGraphics;RasterGraphics;GTK;
+Categories=GTK;GNOME;Graphics;RasterGraphics;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=GIMP
 StartupNotify=true
