diff -Nru gimp-1.1.16/plug-ins/perl/Makefile.PL gimp-1.1.16.new/plug-ins/perl/Makefile.PL
--- gimp-1.1.16/plug-ins/perl/Makefile.PL	Mon Jan 31 04:48:26 2000
+++ gimp-1.1.16.new/plug-ins/perl/Makefile.PL	Fri Feb  4 18:47:16 2000
@@ -205,7 +205,7 @@
 	\$(CHMOD) 755 * ; \\
 	\$(MY_FIXIN) * ; \\
 	for plugin in * ; do \\
-	   $GT \"\$\$plugin\" $GT2 ; \\
+	   $GT \"\$\$plugin\" \$(DESTDIR)$GT2 ; \\
 	done
 	\$(RM_RF) inst-temp
 
@@ -228,9 +228,9 @@
       for (<po/*.po>) {
          my($po)=m!/(.*)\.po$!; #/# for brainy vim
          print " $po";
-         $postamble .= "	\$(MKPATH) \$(datadir)/locale/$po/LC_MESSAGES\n";
-         $postamble .= "	-\$(MSGFMT) -o \$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo po/$po.po\n";
-         $postamble .= "	-\$(CHMOD) 644 \$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n";
+         $postamble .= "	\$(MKPATH) \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES\n";
+         $postamble .= "	-\$(MSGFMT) -o \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo po/$po.po\n";
+         $postamble .= "	-\$(CHMOD) 644 \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n";
       }
    } else {
       print " skipped";
diff -Nru gimp-1.1.16/po/Makefile.in.in gimp-1.1.16.new/po/Makefile.in.in
--- gimp-1.1.16/po/Makefile.in.in	Thu Feb  3 03:10:27 2000
+++ gimp-1.1.16.new/po/Makefile.in.in	Fri Feb  4 18:47:57 2000
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
diff -Nru gimp-1.1.16/po-libgimp/Makefile.in.in gimp-1.1.16.new/po-libgimp/Makefile.in.in
--- gimp-1.1.16/po-libgimp/Makefile.in.in	Thu Jan 27 11:46:58 2000
+++ gimp-1.1.16.new/po-libgimp/Makefile.in.in	Fri Feb  4 18:48:09 2000
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
