diff -urN gimp-1.1.19/plug-ins/perl/Makefile.PL gimp-1.1.19.new/plug-ins/perl/Makefile.PL
--- gimp-1.1.19/plug-ins/perl/Makefile.PL	Tue Mar 28 15:30:27 2000
+++ gimp-1.1.19.new/plug-ins/perl/Makefile.PL	Sat Apr  1 22:16:04 2000
@@ -219,7 +219,7 @@
 	\$(CHMOD) 755 * ; \\
 	\$(MY_FIXIN) * ; \\
 	for plugin in * ; do \\
-	   $GT \"\$\$plugin\" $GT2 ; \\
+	   $GT \"\$\$plugin\" \$(DESTDIR)$GT2 ; \\
 	done
 	\$(RM_RF) inst-temp
 
diff -urN gimp-1.1.19/plug-ins/perl/po/Makefile.PL gimp-1.1.19.new/plug-ins/perl/po/Makefile.PL
--- gimp-1.1.19/plug-ins/perl/po/Makefile.PL	Thu Mar 30 07:14:14 2000
+++ gimp-1.1.19.new/plug-ins/perl/po/Makefile.PL	Sat Apr  1 22:18:03 2000
@@ -7,10 +7,10 @@
    for (<*.po>) {
       my ($po) = /(.*)\.po$/;
       print " $po";
-      $install .= "	@-\$(MKPATH) \$(datadir)/locale/$po/LC_MESSAGES\n".
-                  "	-\$(CP) $po.gmo \$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n".
-                  "	@-\$(CHMOD) 644 \$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n";
-      $uninstall .= "	\$(RM_F) \$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n";
+      $install .= "	@-\$(MKPATH) \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES\n".
+                  "	-\$(CP) $po.gmo \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n".
+                  "	@-\$(CHMOD) 644 \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n";
+      $uninstall .= "	\$(RM_F) \$(DESTDIR)\$(datadir)/locale/$po/LC_MESSAGES/gimp-perl.mo\n";
       $mofiles   .= " $po.gmo";
    }
 } else {
diff -urN gimp-1.1.19/po/Makefile.in.in gimp-1.1.19.new/po/Makefile.in.in
--- gimp-1.1.19/po/Makefile.in.in	Wed Mar 29 12:55:08 2000
+++ gimp-1.1.19.new/po/Makefile.in.in	Sat Apr  1 22:16:04 2000
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
diff -urN gimp-1.1.19/po-libgimp/Makefile.in.in gimp-1.1.19.new/po-libgimp/Makefile.in.in
--- gimp-1.1.19/po-libgimp/Makefile.in.in	Thu Jan 27 11:46:58 2000
+++ gimp-1.1.19.new/po-libgimp/Makefile.in.in	Sat Apr  1 22:16:04 2000
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
diff -urN gimp-1.1.19/po-script-fu/Makefile.in.in gimp-1.1.19.new/po-script-fu/Makefile.in.in
--- gimp-1.1.19/po-script-fu/Makefile.in.in	Tue Mar 28 23:07:48 2000
+++ gimp-1.1.19.new/po-script-fu/Makefile.in.in	Sat Apr  1 22:16:04 2000
@@ -130,7 +130,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
