diff -urN gimp-2.0.6.orig/plug-ins/bmp/bmpread.c gimp-2.0.6/plug-ins/bmp/bmpread.c
--- gimp-2.0.6.orig/plug-ins/bmp/bmpread.c	2004-11-24 15:10:15.000000000 +0000
+++ gimp-2.0.6/plug-ins/bmp/bmpread.c	2004-11-24 15:14:36.187843920 +0000
@@ -457,7 +457,7 @@
       if (grey)
         {
           base_type = GIMP_GRAY;
-          image_type = GIMP_RGB_IMAGE;
+          image_type = GIMP_GRAY_IMAGE;
         }
       else
         {
