--- gimp-2.10.10/configure.ac.orig	2019-04-07 13:38:49.000000000 +0200
+++ gimp-2.10.10/configure.ac	2019-06-16 08:27:35.816441998 +0200
@@ -1956,8 +1956,6 @@
                   [add_deps_error([poppler-glib >= poppler_required_version])])
 # All modern distributions have at least this version and CJK/Cyrillic
 # support should not be considered as "second-zone citizen".
-PKG_CHECK_MODULES(POPPLER_DATA, [poppler-data >= poppler_data_required_version],,
-                  [add_deps_error([poppler-data >= poppler_data_required_version])])
 
 # MIME_TYPES is for the formats we can open as used for instance in the
 # desktop file. Even if cairo-pdf (used for PDF export) is not
