diff -aurN gimp-2.2.0.orig/plug-ins/pygimp/Makefile.am gimp-2.2.0/plug-ins/pygimp/Makefile.am
--- gimp-2.2.0.orig/plug-ins/pygimp/Makefile.am	2004-11-23 03:35:54.000000000 +0100
+++ gimp-2.2.0/plug-ins/pygimp/Makefile.am	2004-12-19 19:48:23.534364736 +0100
@@ -71,4 +71,6 @@
 snap:
 	$(MAKE) dist distdir=$(PACKAGE)-SNAP-`date +"%Y%m%d"`
 
-EXTRA_DIST = makefile.msc
+EXTRA_DIST = \
+	$(pygimp_DATA) \
+	makefile.msc
