diff -ruN gimp-2.2-pre2./data/misc/gimp.desktop.in.in gimp-2.2-pre2/data/misc/gimp.desktop.in.in
--- gimp-2.2-pre2./data/misc/gimp.desktop.in.in	2004-11-26 19:30:03.000000000 +0100
+++ gimp-2.2-pre2/data/misc/gimp.desktop.in.in	2004-11-26 19:36:28.124479016 +0100
@@ -9,7 +9,7 @@
 TryExec=gimp-@GIMP_APP_VERSION@
 Icon=@gimpdatadir@/images/@GIMP_DESKTOP_ICON@
 Terminal=false
-Categories=Application;Graphics;2DGraphics;RasterGraphics;
+Categories=GTK;GNOME;Graphics;RasterGraphics;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=GIMP
 StartupNotify=true
