diff -urN gimp-2.0.6.orig/data/misc/gimp.desktop.in.in gimp-2.0.6/data/misc/gimp.desktop.in.in
--- gimp-2.0.6.orig/data/misc/gimp.desktop.in.in	2004-11-17 12:57:39.774841136 +0100
+++ gimp-2.0.6/data/misc/gimp.desktop.in.in	2004-11-17 13:10:53.398192048 +0100
@@ -7,7 +7,7 @@
 Icon=@gimpdatadir@/images/@GIMP_DESKTOP_ICON@
 Terminal=false
 Type=Application
-Categories=Application;Graphics;RasterGraphics;
+Categories=GTK;GNOME;Graphics;RasterGraphics;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=GIMP
 StartupNotify=true
